frontend/src/routes/Applications/ApplicationDetails/ApplicationTopology/model/topologyAppSet.ts

```
@@ -241,14 +250,17 @@ export async function getAppSetTopology(
       })
     }
-    if (!isApplicationFiltered) {
-      // Collect resource types
+    // Concatenate all resources from all clusters for this app (each resource gets its cluster)
+    const allResourcesForApp: ResourceItem[] = []
+    Object.entries(clusterMap).forEach(([clusterName, resources]) => {
+      if (activeClusters?.length && !activeClusters.includes(clusterName)) {
+        return
+      }
       const types = getResourceTypes(resources as Record<string, unknown>[])
       types.forEach((type) => allApplicationTypes.add(type))
```

Biome (2.4.6)

[error] This callback passed to forEach() iterable method should not return a value.

(lint/suspicious/useIterableCallbackReturn)